src/back/spirv/parser.rs

```diff
@@ -0,0 +1,1545 @@
+/*! Standard Portable Intermediate Representation (SPIR-V) backend !*/
+use super::layout::{Instruction, Module};
+use crate::FastHashMap;
+use spirv::*;
+
+trait LookupHelper<T> {
+    type Target;
+    fn lookup_id(&self, handle: &crate::Handle<T>) -> Option<Word>;
+    fn lookup_handle(&self, word: &Word) -> Option<crate::Handle<T>>;
+}
+
+impl<T> LookupHelper<T> for FastHashMap<Word, crate::Handle<T>> {
+    type Target = T;
+
+    fn lookup_id(&self, handle: &crate::Handle<T>) -> Option<Word> {
+        let mut word = None;
+        for (k, v) in self.iter() {
```

Comment:
functional style would be better here, i.e. `self.iter().find(...).map(...)`

Reply:
Agreed, but this part is still a wonky way to retrieve these values. I am already planning of having other lookups. I will discuss this with you soon.

Reply:
Code is removed